torchao/prototype/mx_formats/mx_subclass.py

```diff
@@ -0,0 +1,124 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD 3-Clause license found in the
+# LICENSE file in the root directory of this source tree.
+
+import types
+from dataclasses import dataclass
+from typing import Optional
+
+import torch
+
+import torchao
+from torchao.core.config import AOBaseConfig
+from torchao.prototype.mx_formats import (
+    MXGemmKernelChoice,
+)
+from torchao.prototype.mx_formats.config import (
+    _validate_elem_dtype,
+    _validate_gemm_kernel_choice,
+)
+from torchao.prototype.mx_formats.mx_tensor import MXTensor
+from torchao.quantization.quant_api import to_linear_activation_quantized
+from torchao.quantization.transform_module import (
+    register_quantize_module_handler,
+)
+from torchao.utils import TORCH_VERSION_AT_LEAST_2_5, is_sm_at_least_100
+
+
+# Note: This API is extra prototype and will change in the future
+@dataclass
+class MXFPConfig(AOBaseConfig):
+    block_size: int = 32
+
+    # Dtypes for Input and Weights
+    activation_dtype: torch.dtype = torch.float8_e4m3fn
+    weight_dtype: torch.dtype = torch.float8_e4m3fn
+
+    # Which kernel to run for mm
+    gemm_kernel_choice: MXGemmKernelChoice = MXGemmKernelChoice.CUBLAS
+
+    # Set some magic perf settings
+    set_inductor_config: bool = True
```

Comment:
can we do this outside the API instead of magically setting here?  OK to punt, but long term IMO this shouldn't be here.

Reply:
Ill flip to default by now, I think that this is mostly impactful for weight only methods and not so much for dynamic